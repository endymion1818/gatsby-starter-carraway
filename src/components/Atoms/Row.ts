import styled from 'styled-components'

export interface IRowProps {
  size: number
  breakpoint?: string
}

const Row = styled.div<IRowProps>`
  @media (min-width: ${({ breakpoint }) => breakpoint}) {
    display: grid;
    grid-template-columns: repeat(${({ size }) => (size ? size : 100)}, 1fr);
    grid-gap: 15px;
  }
  /* IE fallback to float since Grid version is older & likely */
  @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    display: block !important;

    > * {
      display: block !important;
      float: left;
      width: ${({ size }) => (size <= 1 ? 100 : (size / 10) * 100)}%;

      &:not(:first-child) {
        margin-left: 1rem;
      }
      &:not(:last-child) {
        margin-right: 1rem;
      }
    }
    /* clear floats */
    &:after {
      content: '';
      display: block;
      clear: both;
    }
  }
`

export default Row
