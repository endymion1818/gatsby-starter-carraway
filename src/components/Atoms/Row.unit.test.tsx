import { cleanup, render } from '@testing-library/react'
import React from 'react'
import Row from './Row'

describe('<Row />', () => {
  beforeEach(cleanup)
  it('Should render', () => {
    const { container } = render(<Row />)
    expect(container).toMatchSnapshot()
  })
})
