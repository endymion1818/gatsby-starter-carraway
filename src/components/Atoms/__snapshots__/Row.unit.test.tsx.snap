// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> Should render 1`] = `
@media (min-width:) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(100,1fr);
    grid-gap: 15px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c0 {
    display: block !important;
  }

  .c0 > * {
    display: block !important;
    float: left;
    width: NaN%;
  }

  .c0 > *:not(:first-child) {
    margin-left: 1rem;
  }

  .c0 > *:not(:last-child) {
    margin-right: 1rem;
  }

  .c0:after {
    content: '';
    display: block;
    clear: both;
  }
}

<div>
  <div
    class="c0"
  />
</div>
`;
